src/Interpreters/DatabaseCatalog.cpp

```diff
@@ -407,35 +506,39 @@ bool DatabaseCatalog::isPredefinedTable(const StorageID & table_id) const
 
 void DatabaseCatalog::assertDatabaseExists(const String & database_name) const
 {
-    std::lock_guard lock{databases_mutex};
-    assertDatabaseExistsUnlocked(database_name);
+    DatabasePtr db;
+    {
+        std::lock_guard lock{databases_mutex};
+        assert(!database_name.empty());
+        db = databases.find(database_name)->second;
+    }
```

Comment:
this will not work - what if database_name doesn't exist in databases? it will dereference 'end' iterator...
previous variant was completely fine except my question regarding "default" database...